torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -2508,6 +2540,24 @@ void InitXlaModuleBindings(py::module m) {
                          "without a data handle or an IR.";
         });
 
+  // from an XLA tensor to a dlpack tensor.
+  m.def("_to_dlpack", [](const at::Tensor& input) -> py::handle {
+    DLManagedTensor* dlMTensor;
+    {
+      NoGilSection nogil;
+      dlMTensor = torch_xla::toDLPack(input);
+    }
+    // return py::reinterpret_steal<py::object>(PyCapsule_New(dlMTensor,
+    // "dltensor", dlPack_Capsule_Destructor));
+    return PyCapsule_New(dlMTensor, "dltensor", dlPack_Capsule_Destructor);
+  });
+  // m.def("_to_dlpack", &tensor_toDLPack, ""); //
+
+  // from a dlpack tensor to an XLA tensor
+  m.def("_from_dlpack", [](py::handle ext_data) -> at::Tensor {
+    return tensor_fromDLPack(ext_data.ptr());
+  });
+
```

Comment:
Could you leave a comment for each of them, stating what are the assumptions you made when creating those functions? e.g. before calling `_from_dlpack(ext_data)`, we should synchronize the CUDA device if `ext_data` is the result of a CUDA computation. Same thing for `_to_dlpack(input)`.

Reply:
Added it.